app/policies/bgs_policy.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+BGSPolicy = Struct.new(:user, :bgs) do
+  def access?
+    if user.icn.present? && user.ssn.present? && user.participant_id.present?
+      log_success
```

Comment:
I think this method is clearer without the private methods
```
  def access?
    if user.icn.present? && user.ssn.present? && user.participant_id.present?
      StatsD.increment('api.bgs.policy.success') if user.loa3?
      true
    else
      StatsD.increment('api.bgs.policy.failure') if user.loa3?
      false
    end
  end
```
I'm worried the return values for `access?` aren't clear because they are buried in the private methods

Reply:
Cool with me.

Just a FWIW, I 'borrowed heavily' from this file that uses the private method approach. Would you like to maintain parody or have me make the change you requested?

https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/policies/evss_policy.rb

Reply:
@erluti I made your requested changes but code climate wasn't a fan. Not sure how to proceed 🤔 .

![image](https://user-images.githubusercontent.com/4297274/84926426-60476d00-b099-11ea-8ec4-7921366399c3.png)

Reply:
I think writing clear code is more important than matching code somewhere else.
I tested it against the linter, I didn't think about CC.
My first thought is that this method basically returns the value of the if-statement, so I think pulling that back might help.  And then have a method just for the statsd logging
```
def access?
  accessible = user.icn.present? && user.ssn.present? && user.participant_id.present?
  increment_statsd_for_loa_user(accessible)
  accessible
end

def increment_statsd_for_loa_user(accessible)
  return unless user.loa3?
  if accessible
    StatsD.increment('api.bgs.policy.success')
  else
    StatsD.increment('api.bgs.policy.failure')
  end
end
```

Reply:
Looks like my suggestion hit a linting issue.  I'm also surprised it scored `increment_statsd_for_loa_user` as a 6 for cognitive complexity.  I'm okay with that if you are, but the linting issue will block the pipeline.

Reply:
> thanks for responding to my feedback

@erluti Thank you for suggesting them. I didn't see your approval until after I made a change to get past code climate. I'm past CC now. I'll work on the security issue that popped up.

Link for your convenience:

https://github.com/department-of-veterans-affairs/vets-api/pull/4331/commits/cc4ce05bf22ec33aac7b2ad825374840236f8918